src/scribe.js

```diff
@@ -40,32 +57,67 @@ define([
      */
 
     var isComposing = false;
-    var self = this;
+    var scribe = this;
+    var shouldTransact = true;
+    var timer = 0;
+    var throttleInterval = scribe.options.undo.interval || 250;
 
     // For IE11
-    if (/Trident/.test(navigator.userAgent)) {
+    if (isIE) {
       var handler = {
         handleEvent: function(e) {
+          configureZeroState(scribe);
+
           if (isComposing) return;
 
-          if (e.type === 'compositionstart') {
-            isComposing = true;
-            return;
-          } else if (e.type === 'compositionend') {
-            isComposing = false;
-            self.transactionManager.run();
-          } else {
-            self.transactionManager.run();
-          }
+          switch (e.type) {
+						case 'compositionstart':
+						  isComposing = true;
+							break;
+						case 'compositionend':
+						  isComposing = false;
+							break;
+						case 'keydown':
+						  if (e.which === 32 || e.which === 13) {
+                // prevent unnecessary history entries when pressing space in rappid succession
+                if (shouldTransact) {
+								  scribe.transactionManager.run();
+                  shouldTransact = false;
+                }
+                clearTimeout(timer);
+                timer = setTimeout(function() {
+                  shouldTransact = true;
+                }, throttleInterval);
+							}
+							break;
+						case 'cut':
+						  scribe.transactionManager.run();
+							break;
+					}
         }
       };
 
-      ['compositionstart', 'compositionend', 'keydown', 'cut', 'paste'].forEach(function(e) {
-        this.el.addEventListener(e, handler, false);
-      }, this);
+      ['compositionstart', 'compositionend', 'keydown', 'cut'].forEach(function(e) {
+        scribe.el.addEventListener(e, handler, false);
+      });
+
+      var onPaste = function() {
+        scribe.transactionManager.run();
+      };
+
+			scribe.on('paste', onPaste);
+
+      scribe.on('destroy', function() {
+        scribe.off('paste', onPaste);
+        ['compositionstart', 'compositionend', 'keydown', 'cut'].forEach(function(e) {
```

Comment:
Should we keep this array of events types as a variable at the top since we're using them in two places?
Just in case, so we don't need to keep them in sync manually.